python/ribasim/ribasim/config.py

```diff
@@ -0,0 +1,117 @@
+# generated by datamodel-codegen:
+#   filename:  Config.schema.json
+#   timestamp: 2023-09-12T09:11:45+00:00
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import List, Optional, Union
+
+from pydantic import BaseModel, Field
+
+
+class Output(BaseModel):
+    compression: Optional[str] = Field("zstd", description="compression")
+    basin: Optional[str] = Field("output/basin.arrow", description="basin")
+    flow: Optional[str] = Field("output/flow.arrow", description="flow")
+    control: Optional[str] = Field("output/control.arrow", description="control")
+    outstate: Optional[str] = Field(None, description="outstate")
+    compression_level: Optional[int] = Field(6, description="compression_level")
+
+
+class LevelBoundary(BaseModel):
+    time: Optional[str] = Field(None, description="time")
+    static: Optional[str] = Field(None, description="static")
+
+
+class Pump(BaseModel):
+    static: Optional[str] = Field(None, description="static")
+
+
+class DiscreteControl(BaseModel):
+    logic: Optional[str] = Field(None, description="logic")
+    condition: Optional[str] = Field(None, description="condition")
+
+
+class Solver(BaseModel):
+    reltol: Optional[float] = Field(0.001, description="reltol")
+    saveat: Optional[Union[List[float], float]] = Field([], description="saveat")
+    maxiters: Optional[int] = Field(1000000000, description="maxiters")
+    autodiff: Optional[bool] = Field(True, description="autodiff")
+    adaptive: Optional[bool] = Field(True, description="adaptive")
+    algorithm: Optional[str] = Field("QNDF", description="algorithm")
```

Comment:
Is `Optional` still needed? Before when no value was given it was None, but now that it has defaults, isn't it just `str`?

Reply:
This file is automatically generated, we allow nothing in Julia, thus this is Optional here.

Reply:
We don't directly allow nothing right? Or is that because the solver section itself is optional?

https://github.com/Deltares/Ribasim/blob/f503acb8674a6c59153298f655d036d82e1d2107/core/src/config.jl#L73

But so you agree that not having Optional here does not serve any purpose?